apps/cli/src/commands/doctor.ts

```diff
@@ -11,102 +11,104 @@ export default class DoctorCommand extends Command {
-        return semver.gte(version, DoctorCommand.MINIMUM_DOCKER_VERSION);
-    }
+    private async checkDocker(): Promise<true | never> {
+        try {
+            let { stdout: dockerVersion } = await execa("docker", [
+                "version",
+                "--format",
+                "{{json .Client.Version}}",
+            ]);
 
-    private static isDockerComposeVersionValid(version: string): boolean {
-        const v = semver.coerce(version);
-        return (
-            v !== null &&
-            semver.gte(v, DoctorCommand.MINIMUM_DOCKER_COMPOSE_VERSION)
-        );
-    }
+            const v = semver.coerce(dockerVersion);
+            if (
+                v !== null &&
+                !semver.gte(v, DoctorCommand.MINIMUM_DOCKER_VERSION)
+            ) {
+                throw new Error(
+                    `Unsupported Docker version. Minimum required version is ${DoctorCommand.MINIMUM_DOCKER_VERSION}. Installed version is ${v}.`,
+                );
+            }
+        } catch (e: unknown) {
+            if ((e as any).code == "ENOENT") {
+                throw new Error("Docker is required but not installed.");
+            } else {
+                throw e;
+            }
+        }
 
-    private static isBuildxRiscv64Supported(buildxOutput: string): boolean {
-        return buildxOutput.includes("riscv64");
+        return true;
     }
 
-    public async run() {
+    private async checkCompose(): Promise<true | never> {
         try {
-            try {
-                const dockerVersion = await DoctorCommand.checkDockerVersion();
+            const { stdout: dockerComposeVersion } = await execa("docker", [
+                "compose",
+                "version",
+                "--short",
+            ]);
 
-                // Check Docker version
-                if (!DoctorCommand.isDockerVersionValid(dockerVersion)) {
-                    throw new Error(
-                        `Unsupported Docker version. Minimum required version is ${DoctorCommand.MINIMUM_DOCKER_VERSION}. Installed version is ${dockerVersion}.`,
-                    );
-                }
-            } catch (error: unknown) {
-                if (error instanceof Error) {
-                    this.error(
-                        "Docker is required but not installed or the command execution failed. Please refer to the Docker documentation for installation instructions: https://docs.docker.com/get-docker/",
-                    );
-                }
+            const v = semver.coerce(dockerComposeVersion);
+            if (
+                v !== null &&
+                !semver.gte(v, DoctorCommand.MINIMUM_DOCKER_COMPOSE_VERSION)
+            ) {
+                throw new Error(
+                    `Unsupported Docker Compose version. Minimum required version is ${DoctorCommand.MINIMUM_DOCKER_COMPOSE_VERSION}. Installed version is ${v}.`,
+                );
             }
```

Comment:
When I run doctor on a system without docker, It can't catch the error and show the message properly: 

```
    Error: Command failed with ENOENT: docker compose version --short
    spawn docker ENOENT
    Code: ENOENT
```

Reply:
hmm, so we should only check for compose is docker is installed first

```patch
diff --git a/apps/cli/src/commands/doctor.ts b/apps/cli/src/commands/doctor.ts
index d37a624c..9498475b 100644
--- a/apps/cli/src/commands/doctor.ts
+++ b/apps/cli/src/commands/doctor.ts
@@ -102,9 +102,10 @@ export default class DoctorCommand extends Command {
 
     public async run(): Promise<void> {
         try {
-            await this.checkDocker();
-            await this.checkCompose();
-            await this.checkBuildx();
+            if (await this.checkDocker()) {
+                await this.checkCompose();
+                await this.checkBuildx();
+            }
         } catch (e: unknown) {
             this.error(e as Error);
         }

```

Reply:
Already checked and I've got the same error: 

```
    Error: Command failed with ENOENT: docker compose version --short
    spawn docker ENOENT
    Code: ENOENT

```